src/sdk/namespace_fs.js

```diff
@@ -1111,10 +1118,10 @@ class NamespaceFS {
             upload_params = await this._start_upload(fs_context, object_sdk, file_path, params, open_mode);
 
             if (!params.copy_source || upload_params.copy_res === copy_status_enum.FALLBACK) {
-            // TODO: Take up only as much as we need (requires fine-tune of the semaphore inside the _upload_stream)
-            // Currently we are taking config.NSFS_BUF_SIZE for any sized upload (1KB upload will take a full buffer from semaphore)
-                const upload_res = await buffers_pool_sem.surround_count(
-                    config.NSFS_BUF_SIZE, async () => this._upload_stream(upload_params));
+            // We are taking the buffer size closest to the sized upload
+                const bp = multi_buffer_pool.get_buffers_pool(params.size >= 0 ? params.size : config.NSFS_BUF_SIZE_L);
```

Comment:
we want to avoid reading the config dynamically and instead use the largest pool.
use get_buffers_pool(params.size) and handled it inside the call to return the largest pool in that case.